src/libs/Middleware/Reauthentication.ts

```diff
@@ -0,0 +1,109 @@
+            }
+
+            // There are some API requests that should not be retried when there is an auth failure like
+            // creating and deleting logins. In those cases, they should handle the original response instead
+            // of the new response created by handleExpiredAuthToken.
+            const shouldRetry = request?.data?.shouldRetry;
+            const apiRequestType = request?.data?.apiRequestType;
+
+            // For the SignInWithShortLivedAuthToken command, if the short token expires, the server returns a 407 error,
+            // and credentials are still empty at this time, which causes reauthenticate to throw an error (requireParameters),
+            // and the subsequent SaveResponseInOnyx also cannot be executed, so we need this parameter to skip the reauthentication logic.
+            const skipReauthentication = request?.data?.skipReauthentication;
+            if ((!shouldRetry && !apiRequestType) || skipReauthentication) {
+                if (isFromSequentialQueue) {
+                    return data;
+                }
+
+                if (request.resolve) {
+                    request.resolve(data);
+                }
+
+                return data;
+            }
+
+            // We are already authenticating and using the DeprecatedAPI so we will replay the request
+            if (!apiRequestType && NetworkStore.isAuthenticating()) {
+                MainQueue.replay(request);
+                return data;
+            }
+
+            return reauthenticate(request?.commandName)
+                .then((authenticateResponse) => {
+                    if (isFromSequentialQueue || apiRequestType === CONST.API_REQUEST_TYPE.MAKE_REQUEST_WITH_SIDE_EFFECTS) {
+                        return Request.processWithMiddleware(request, isFromSequentialQueue);
+                    }
+
+                    if (apiRequestType === CONST.API_REQUEST_TYPE.READ) {
+                        NetworkConnection.triggerReconnectionCallbacks('read request made with expired authToken');
+                        return Promise.resolve();
+                    }
+
+                    MainQueue.replay(request);
+                    return authenticateResponse;
+                })
+                .catch(() => {
+                    if (isFromSequentialQueue || apiRequestType) {
+                        throw new Error('Failed to reauthenticate');
+                    }
+
+                    // If we make it here, then our reauthenticate request could not be made due to a networking issue. The original request can be retried safely.
+                    MainQueue.replay(request);
+                });
+        }
+
+        if (isFromSequentialQueue) {
+            return data;
+        }
+
+        if (request.resolve) {
+            request.resolve(data);
+        }
+
+        // Return response data so we can chain the response with the following middlewares.
+        return data;
+    });
```

Comment:
Why did you remove the entire `catch` clause?

Reply:
My bad, good catch @fabioh8010